tests/test_pvc_concurrent_deletion_creation.py

```diff
@@ -0,0 +1,223 @@
+    )
+
+    def async_communicate():
+        stdout, stderr = p.communicate()
+        retcode = p.returncode
+        return retcode, stdout, stderr
+
+    p.async_communicate = async_communicate
+    return p
+
+
+def verify_pvc_bound(pvc_name):
+    """
+    Verify that the status of PVC is 'Bound'
+
+    Args:
+        pvc_name (str): Name of PVC
+
+    Returns:
+        True if PVC is in Bound state, False if PVC does not exists.
+    """
+    try:
+        pvc_info = PVC.get(pvc_name)
+        assert pvc_info['status']['phase'] == "Bound", (f'PVC {pvc_name} is '
+                                                        f'not in Bound state')
+    except exceptions.CommandFailed:
+        log.info(f"PVC {pvc_name} doesn't exist")
+        return False
+    return True
+
+
+def verify_pvc_not_exists(pvc_name):
+    """
+    Ensure that the pvc does not exists
+
+    Args:
+        pvc_name (str): Name of PVC
+
+    Returns:
+        True if PVC does not exists, False otherwise.
+    """
+    try:
+        PVC.get(pvc_name)
+        return False
+    except exceptions.CommandFailed as exp:
+        assert "not found" in str(exp)
+        log.info(f'Expected: PVC {pvc_name} does not exists ')
+    return True
+
+
+@tier1
+@pytest.mark.usefixtures(
+    test_fixture.__name__,
+)
+class TestMultiplePvcConcurrentDeletionCreation(ManageTest):
+    """
+    Test to verify concurrent creation and deletion of multiple PVCs
+    """
+    def test_multiple_pvc_concurrent_creation_deletion(self):
+        """
+        To exercise resource creation
+        """
+        sc_parms = {'sc_name': 'test-sc'}
+        pvc_base_name = 'test-pvc'
+        number_of_pvc = 500
+
+        # Create storage class
+        create_storage_class(**sc_parms)
+
+        # Create 500 PVCs
+        create_multiple_pvc(pvc_base_name, number_of_pvc)
+
+        # Verify PVCs exists
+        for count in range(1, number_of_pvc + 1):
+            pvc_name = f'{pvc_base_name}{count}'
+            assert verify_pvc_bound(pvc_name)
+
+        # Start deleting 500 PVCs
+        command = (f'for i in `seq 1 {number_of_pvc}`;do oc '
+                   f'delete pvc {pvc_base_name}$i;done')
+        proc = run_async(command)
+        assert proc, (f'Failed to execute command for deleting '
+                      f'{number_of_pvc} PVCs')
+
+        # Create another 500 PVCs
+        pvc_base_name_new = 'test-pvc-re'
+        create_multiple_pvc(pvc_base_name_new, number_of_pvc)
+
+        # Verify PVCs exists
+        for count in range(1, number_of_pvc + 1):
+            pvc_name = f'{pvc_base_name_new}{count}'
+            assert verify_pvc_bound(pvc_name)
+
+        # Verify command to delete PVCs
+        ret, out, err = proc.async_communicate()
+        log.info(f'Return values of command: {command}.'
+                 f'\nretcode:{ret}\nstdout:{out}\nstderr:{err}')
+        assert not ret, "Deletion of PVCs failed"
+
```

Comment:
@jilju, along with validating if pvc name doesn't exists(its deleted successfully), we may also need to validate if the corresponding pv also got deleted.